lib/blockchain/chain.js

```diff
@@ -3161,6 +3161,45 @@ class Chain extends AsyncEmitter {
     return state;
   }
 
+  /**
+   * Get signalling statistics for BIP9/versionbits soft fork
+   * @param {ChainEntry} prev - Previous chain entry.
+   * @param {String} id - Deployment id.
+   * @returns {Promise} - Returns JSON object.
+   */
+
+  async getBIP9Stats(prev, deployment) {
```

Comment:
What is the expected behavior of this function when the soft fork is not in the `STARTED` state? Since this is a library function, I think we should handle all possible inputs.

Reply:
Hm, what do you suggest? Outside of `STARTED` the statistics are totally meaningless, but they would still be accurate (reporting the number of blocks with a certain bit set). The exception would be `possible` since activations are only possible or not during the `STARTED` phase.

We could make another call to `getState()` and assert that we are `STARTED` ?

Reply:
I think throwing an error would be fine.

Reply:
Added a commit to catch this state and throw. I'm not sure if this is the best method, since it calls `getState()` again, meaning this:

`while((block.height + 1) % window !== 0)`

is run three times for each `rpc getblockchaininfo` call (to get the state in the rpc module, to get it again in the chain module, and then finally to actually count the statistics(

Reply:
@pinheadmz where does `getState` run the while loop you are describing?

Reply:
Regardless, the while loop is bounded by `window`, so I'm not sure that it will make a big difference. If it does we can pass `state` to `getBIP9Stats`.

Reply:
I mean, `getState` itself iterates through up to 2016 to check the state. That will happen twice, and then `getBIP9Stats` does it again to actually count the signalling bits .